lib/analist/annotations.rb

```diff
@@ -6,11 +6,22 @@ class Annotation
 
     def initialize(receiver_type, args_types, return_type, hint: nil)
       @receiver_type = receiver_type.is_a?(Hash) ? receiver_type : { type: receiver_type }
-      @args_types = args_types
+      @args_types = self.class.initialize_args_types(args_types)
       @return_type = return_type.is_a?(Hash) ? return_type : { type: return_type }
       @hint = hint
     end
 
+    def self.initialize_args_types(args_types)
+      if args_types.is_a?(Set)
+        return Set.new(args_types.map do |set|
+          set.map do |arg_type|
```

Comment:
You might be able to change this to a recursive function call as it is basically the case handled by the loop below.

Reply:
Good catch!